### PR TITLE
Fix index out of bounds during parser error handling

### PR DESCRIPTION
**Pull request by martint**
Stop once we reach the end of the token stream.

### CODE REVIEW

**Review by dain — APPROVED** [review]

File: presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java [hunk 1]
```diff
@@ -293,7 +293,7 @@ else if (transition instanceof WildcardTransition) {
                             labels = labels.complement(IntervalSet.of(Token.MIN_USER_TOKEN_TYPE, atn.maxTokenType));
                         }
 
-                        if (labels.contains(currentToken)) {
+                        if (labels.contains(currentToken) && tokenIndex < stream.size() - 1) {
```
Comment (findepi) [inline presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java]: consider adding an explanation why `labels` contains EOF.
Normally EOF token should be good enough of the sentinel.

Comment (martint) [reply presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java]: It's actually more nuanced than that. In the specific example I added, the token stream that's handed to the error handler hasn't read tokens all the way to the end of the stream, so the stream ends with a non-EOF token. I could force it to keep going, but if Antlr decided the error is somewhere within the currently loaded set of tokens, I don't think it buys us much to try to keep simulating.

Comment (findepi) [reply presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java]: Makes sense. Would be good to have a trace of this knowledge in the code comment.

Comment (martint) [reply presto-parser/src/main/java/io/prestosql/sql/parser/ErrorHandler.java]: Added a comment

**Pull request merged**